api/types/events/oneof.go

```diff
@@ -353,8 +355,28 @@ func ToOneOf(in AuditEvent) (*OneOf, error) {
 		out.Event = &OneOf_RenewableCertificateGenerationMismatch{
 			RenewableCertificateGenerationMismatch: e,
 		}
+	case *Unknown:
+		out.Event = &OneOf_Unknown{
+			Unknown: e,
+		}
 	default:
-		return nil, trace.BadParameter("event type %T is not supported", in)
+		log.Errorf("Attempted to convert dynamic event of unknown type \"%v\" into protobuf event.", in.GetType())
+		unknown := &Unknown{}
+		unknown.Type = "unknown"
+		unknown.Code = "TCC00E"
```

Comment:
note to reviewers: this isn't pretty but is unavoidable in the absence of significant package refactoring due to cyclic dependencies

Reply:
Can you define these in api/types/events then just alias them in lib/events?

```
UnknownCode = events.UnknownCode
```

Reply:
Sure, that at least saves us constant duplication